miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java

```diff
@@ -314,11 +304,16 @@ public ModelAndView editPools(@RequestParam("ids") String poolIds, ModelMap mode
     return bulkEditBackend.edit(poolIds, model);
   }
 
-  private final BulkMergeTableBackend<PoolDto> bulkMergeBackend = new BulkMergeTableBackend<PoolDto>(
-      "pool", PoolDto.class, "Pool", "Pools") {
+  private static class BulkMergePoolsBackend extends BulkTableBackend<PoolDto> {
 
-    @Override
-    protected PoolDto createDtoFromParents(List<Long> parentIds) throws IOException {
+    private final PoolService poolService;
+
+    public BulkMergePoolsBackend(PoolService poolService) {
+      super("pool", PoolDto.class);
+      this.poolService = poolService;
+    }
+
+    protected PoolDto createDtoFromParents(List<Long> parentIds, List<Integer> proportions) throws IOException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 21 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)